GeosCore/tomas_mod.F90

```diff
@@ -1745,19 +1638,28 @@ SUBROUTINE getCondSink(Nko, Mko, spec, CS, sinkfrac, surf_area, &
        Kn      = 2.0 * l_ab / Dpk(k)     !S&Pv2 chapter 12 - Kn for Dahneke correction factor
        beta(k) = ( 1.+Kn )  / ( 1.+2.*Kn*(1.+Kn)/alpha(spec) )   !S&P eqn 11.35
     enddo
-
+    
     ! get condensation sink
     CS = 0.e+0_fp
     surf_area = 0.e+0_fp
     do k=1,ibins
        CS = CS + Dpk(k)*Nko(k)*beta(k)
        surf_area = surf_area+Nko(k)*pi*(Dpk(k)*1.0e+6_fp)**2
     enddo
+    !bc 21/01/2022 - check if divide by zero below -added 2 if 
     do k=1,ibins
-       sinkfrac(k) = Dpk(k)*Nko(k)*beta(k)/CS
+       if (CS > 0.e-0_fp) then
+          sinkfrac(k) = Dpk(k)*Nko(k)*beta(k)/CS
+       else
+          sinkfrac(k) = 0.e-0_fp
+       endif
```

Comment:
We can get rid of the ELSE with:
```F90
    do k=1,ibins
       sinkfrac(k) = 0.0_fp
       if (CS > 0.0_fp) then
          sinkfrac(k) = Dpk(k)*Nko(k)*beta(k)/CS
       endif
    enddo
```

Reply:
This change bas been made in https://github.com/geoschem/geos-chem/pull/2060/commits/2d6ee6630102a6a11b7643d9d4094a12c9fa453c.